src/app/modules/group/components/user-list/user-list.component.html

```diff
@@ -1,45 +1,65 @@
-<p-progressSpinner *ngIf="state === 'fetching'  && members.length == 0"></p-progressSpinner>
-
 <p *ngIf="state === 'error'">Error while loading the group members</p>
 
-<alg-grid #grid
-  [data]="members"
-  [showGear]="false"
-  (sort)="onCustomSort($event)"
-  [loading]="state === 'fetching' && members.length > 0"
+<alg-group-composition-filter (change)="onPolicyChange($event)" [defaultValue]="currentPolicy"></alg-group-composition-filter>
+
+<p-table
+  [columns]="data.columns"
+  [value]="data.rowData"
+  [customSort]="true"
+  sortMode="multiple"
+  [lazy]="true"
+  (onLazyLoad)="onCustomSort($event)"
```

Comment:
I used the lazy loading because the `onCustomSort` function was called just after the table was updated, so it updates the table again, thus creating an infinite loop. I found this solution [here](https://github.com/primefaces/primeng/issues/5163).

Reply:
ok :-/
but so you arg type of `onCustomSort` is not a SortEvent anymore.
From their code I get 
```
 return {
            first: this.first,
            rows: this.rows,
            sortField: this.sortField,
            sortOrder: this.sortOrder,
            filters: this.filters,
            globalFilter: this.filters && this.filters['global'] ? (<FilterMetadata> this.filters['global']).value : null,
            multiSortMeta: this.multiSortMeta
        };
```
By chance, it has `multiSortMeta` as well, but that's a coincidence.

Reply:
Not addressed?

Reply:
fixed, I switched back to using `sortFunction` instead of `onLazyLoad`.
the `onCustomSort` is still called 2 times (once when the user sorts a column, and once after the new sorted data is fetched) but this doesn't create an infinite loop anymore